airflow/models/dag.py

```diff
@@ -594,7 +594,7 @@ def __init__(
         self.timezone: Timezone | FixedTimezone = tz or settings.TIMEZONE
 
         # Apply the timezone we settled on to end_date if it wasn't supplied
-        if "end_date" in self.default_args and self.default_args["end_date"]:
+        if self.default_args.get("end_date"):
```

Comment:
GitHub does not allow me to suggest inline. This can be

```python
if isinstance(end_date := self.default_args.get("end_date"), str):
    self.default_args["end_date"] = timezone.parse(end_date, timezone=self.timezone)
```